.github/workflows/rust.yml

```diff
@@ -1,5 +1,40 @@
 name: Rust
 
+# On Rust, GitHub Actions, and caching
+# ===========
+# Here's a list of things to keep in mind if you find yourself maintaing this
+# CI:
+#
+# https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+#
+# - Always install and select the desired Rust toolchain *before* running
+#   `Swatinem/rust-cache`. This is because the active Rust toolchain is used as
+#   a cache key.
+# - When caching Rust compilation artifacts, keep in mind that different `cargo`
+#   commands will use different profiles
+#   (https://doc.rust-lang.org/cargo/reference/profiles.html). Learn what you
+#   can reuse between one job and another and don't assume two commands will
+#   just share caches without conflicts.
+# - Be extremely aware of cache thrashing a.k.a. churning. GitHub Actions' cache
+#   allows for 10GiB of data which is easily exceeded if not careful.
+#   Sometimes it's better not to cache than cache excessively.
+#   We might even want to disable caching writes for non-default branches
```

Comment:
I think this is a good idea!!!! Cache write only on nightly/stable.

maybe let's do it right away?

Reply:
Sure, let's try! https://github.com/Sovereign-Labs/sovereign-sdk/pull/572/commits/b604d67de446f5a856d1a6e9993e91106394b4a0.